src/Jaeger.Core/SpanContext.cs

```diff
@@ -1,54 +1,132 @@
 using System;
 using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Globalization;
 using OpenTracing;
 
 namespace Jaeger.Core
 {
-    // The SpanContext is used to pass the information needed by other spans so
-    // that they can correctly reference other spans
-    public class SpanContext : IJaegerCoreSpanContext
+    public class SpanContext : ISpanContext
     {
+        internal static readonly ReadOnlyDictionary<string, string> EmptyBaggage = new ReadOnlyDictionary<string, string>(new Dictionary<string, string>());
```

Comment:
Use `IReadOnlyDictionary`. `Dictionary` already implements `IReadOnlyDictionary`. Wrapping in `ReadOnlyDictionary` not needed.